pkg/cmd/pr/create/create.go

```diff
@@ -480,9 +480,14 @@ func NewCreateContext(opts *CreateOptions) (*CreateContext, error) {
 	}
 	client := api.NewClientFromHTTP(httpClient)
 
+	// TODO: consider obtaining remotes from GitClient instead
 	remotes, err := opts.Remotes()
 	if err != nil {
-		return nil, err
+		// When a repo override value is given, ignore errors when fetching git remotes
+		// to support using this command outside of git repos.
+		if opts.RepoOverride == "" {
+			return nil, err
+		}
```

Comment:
Seems a bit off to run `opts.Remotes()` when `opts.RepoOverride` is not a blank string since we know it will fail. What do you think about doing something like this:
```go
	var remotes ghContext.Remotes
	if opts.RepoOverride == "" {
		remotes, err = opts.Remotes()
		if err != nil {
			return nil, err
		}
	}
```

Reply:
That's a good suggestion, but `gh pr create` is an odd command in it that the local git repo still has a purpose even if you've used the `-R/--repo` override.

In other commands, the presence of the `--repo` flag means that the local git repo should not even be considered; but in `gh pr create` the `--repo` flag can be used to change the base repo that the PR will be opened against; however the head branch, branch push status, head repo etc. can still be determined from the local git repo (if one exists). The `pr create` command being an exception like this may be counter-intuitive but is kept for backwards compatibility.